### PR TITLE
Should GetVirtualPath return PathString

### PR DESCRIPTION
**Pull request by ianhong**
@rynowak 

I created a new PR. This is for #31. Since I have 40+ tests to update + probably more from the MVC repo, I'd like to make sure this fix is correct before updating more tests.

Thanks,

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@ianhong__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Youngjune Hong). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla2.msopentech.com.
    </p>

TTYL, MSOTBOT;

File: src/Microsoft.AspNet.Routing/RouteCollection.cs [hunk 1]
```diff
@@ -203,7 +203,7 @@ public virtual VirtualPathData GetVirtualPath(VirtualPathContext context)
                 {
                     return new VirtualPathData(
                         bestPathData.Router,
-                        NormalizeVirtualPath(bestPathData.VirtualPath),
+                        NormalizeVirtualPath(bestPathData.VirtualPath.Value),
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/RouteCollection.cs]: Change `NormalizeVirtualPath` to return a `PathString`

File: src/Microsoft.AspNet.Routing/VirtualPathData.cs [hunk 2]
```diff
@@ -36,7 +37,7 @@ public VirtualPathData([NotNull] IRouter router, string virtualPath)
             IDictionary<string, object> dataTokens)
         {
             Router = router;
-            VirtualPath = virtualPath;
+            VirtualPath = CreatePathString(virtualPath);
```
Comment (rynowak) [inline src/Microsoft.AspNet.Routing/VirtualPathData.cs]: Let's also add a constructor that takes a `PathString`

**Comment by ianhong** [comment]
@rynowak 
I updated per your comments..

Thank you

**Comment by rynowak** [comment]
:shipit: for the design

**Comment by ianhong** [comment]
@rynowak 

Could you take a look at this? I updated those tests, I'm now working on the MVC repo for this change.

**Comment by ianhong** [comment]
@rynowak 
Please ignore this. I'll send another PR. This PR seems just confusing..

**Pull request closed without merging**